File: frontend/src/enginePalette.ts
// enginePalette.ts — client mirror of the server's EnginePalette colour IDS: a seat/team
// index maps to the colour WORD spoken as its identity («Equipo Rojo») and matches the hex
// the server dressed the players in. A new entry here needs its game.color_* word too.

export const ENGINE_PALETTE_NAMES =
	['red', 'blue', 'yellow', 'green', 'purple', 'orange', 'cyan', 'brown'] as const;

/** The colour-word id for a seat/team index ("red" → game.color_red). */
export function teamColorId(index: number): string {
	return ENGINE_PALETTE_NAMES[index % ENGINE_PALETTE_NAMES.length];
}

/** The localized team name for a seat/team index («Equipo Rojo» / "Red team"). */
export function teamDisplayName(
	index: number,
	tSync: (key: string, vars?: Record<string, unknown>) => string,
): string {
	return tSync('game.journey_team', { color: tSync(`game.color_${teamColorId(index)}`) });
}

File: server/Services/Corro/Families/EnginePalette.cs
namespace CorroServer.Services.Corro.Families;

/// <summary>
/// The engine's player-colour palette, assigned in join order by families whose board has no
/// seat colours of its own (track pieces, journey cars). The client mirrors these hexes to
/// spoken colour WORDS ("your colour is #1e88e5" would be meaningless aloud), so a new entry
/// here needs its word on the client too.
/// </summary>
public static class EnginePalette
{
	private static readonly string[] Colors =
		{ "#e53935", "#1e88e5", "#fdd835", "#43a047", "#8e24aa", "#fb8c00", "#00acc1", "#6d4c41" };

	/// <summary>Colour IDS aligned with <see cref="Colors"/>: the client localizes them into
	/// the spoken colour word (game.color_*) — team names ("Equipo Rojo") are built from these.</summary>
	private static readonly string[] Names =
		{ "red", "blue", "yellow", "green", "purple", "orange", "cyan", "brown" };

	public static string ColorFor(int index) => Colors[index % Colors.Length];

	public static string NameFor(int index) => Names[index % Names.Length];
}
